interview-questions.md

```diff
@@ -4,65 +4,64 @@ Answer the following questions. First, without external resources. Challenge you
 
 
 
 3. What is the difference between map/filter and a for loop?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Map/filter are built-in methods for JavaScript, they are prewritten functions that perform the task of iterating through an array more efficiently. A for loop is a broadly useful tool that, while it can be used for iterating through an array, may not be the most effective. 
 
+  Researched answer: Map functions can also be cleaner and easier to read for people reviewing your code at a later date. 
 
 
 4. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: console.log is a developer tool used to see what is going on with the code, return is the actual useful value a function spits out after running its program. 
 
-  Researched answer:
+  Researched answer: Console.log prints information to the console, while return is a call to pass some value back up to where the call was made. 
 
 
 
 5. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: An argument is like the variable in algebra, it is whatever we want to be able to change between each running of the function. 
 
-  Researched answer:
+  Researched answer: The argument is the value that must be provided to obtain the function's result. 
 
 
 
 6. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Proper pair programming technique should involve a driver and a navigator. The driver is typing and dealing with the minutia of the code while the navigator is considering the bigger picture of the coding problem while watching out for bugs or mistakes. 
 
-  Researched answer:
+  Researched answer: The navigator is also critical in brainstorming
 
 
 
 7. What is something we did in class this week you found helpful?  
 
-  Your answer:
+  Your answer: I really enjoyed getting hands-on practice with GitHub, coding challenges, and pair-programming. I found it very helpful to work with others and hear how they think about/solve problems and coming up with creative solutions that play to the strengths of both of our ideas. 
 
 
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
-- Jest
-- Class inheritance
-- React
-- Yarn
-- React State
+- Jest - Jest is a JavaScript testing framework/library that was created by Facebook and is particularly well-suited for React code. 
+- Class inheritance - A way for one class to extend functionality to another class.
+- React - An opensource JavaScript library for creating User Interfaces.
+- Yarn - A JavaScript Package Manager, or a way for developers to share code so they don't have to rewrite the same things. 
+- React State - A built in object on React components that are used to store property values for the component. 
 - this
```

Comment:
All looks good, but I don't see an answer to 'this'. If you weren't sure about it so you left it blank, no problem! Next time i'd just write something there and say "Not sure!" just to let us know :)